### PR TITLE
test: added test github api for runs

### PR DESCRIPTION
**Pull request by Aleksey28**
1. Refactored previous code
2. Installed sinon and proxyquire
3. Created test

### CODE REVIEW

File: index.ts [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import { Config, debugApp, debugConfig } from './src/config.js';
-import esMain from 'es-main';
+// import esMain from 'es-main';
```
Comment (AndreyBelym) [inline index.ts]: Please don't be afraid to remove unnecessary code, since with git it can be easily reverted 😄

File: src/apiGithub.ts [hunk 2]
```diff
@@ -34,16 +34,23 @@ export default class ApiGithub {
...
+
         return got(`${this._baseUrl}${relativePath}`, options);
     }
 
     _handleResponse (response: Response): HandleResponseInterface {
 
-        const result = typeof response.body === 'string' ? JSON.parse(response.body) : '';
+        debugApiGithub(`Response: ${response}`);
+
+        const result = typeof response.body === 'string' ? JSON.parse(response.body) : response.body;
```
Comment (AndreyBelym) [inline src/apiGithub.ts]: It's good for now, but I suggest you to take a closer look at [Type Guards](https://www.typescriptlang.org/docs/handbook/advanced-types.html). You could add more robust type checks with them by using separate interfaces for completed responses and errors.

File: test/config-test.ts [hunk 3]
```diff
@@ -90,4 +101,29 @@ describe('Checking server', () => {
     });
 });
 
-//TODO: write tests for github APIs after migrating to TypeScript
+describe('Checking api GitHub', () => {
+    it('Getting runs should working', async () => {
+        const myGot = sinon.stub();
```
Comment (AndreyBelym) [inline test/config-test.ts:106]: Yep, that's the test I was waiting for 😉

File: index.ts (hunk 1 above)
Comment (AndreyBelym) [reply index.ts]:
```suggestion
```

Comment (AndreyBelym) [reply index.ts]: As an exercise, try to apply all my suggestions at once 😉

File: index.ts [hunk 4]
```diff
@@ -17,7 +17,8 @@ async function runServer (): Promise<void> {
     debugApp(`App "${packageJSON.name}" is listening on port: ${Config.globals.port}`);
 }
 
-if (esMain(import.meta))
+// if (import.meta ? esMain(import.meta) : require.main === module)
```
Comment (AndreyBelym) [inline index.ts]:
```suggestion
```

**Review by AndreyBelym — COMMENTED** [review]
Excellent job, just don't leave code commits behind when making a PR.

**Comment by Aleksey28** [comment]
FPR

**Review by AndreyBelym — APPROVED** [review]
Awesome

**Pull request merged**